scripts/lib/CIME/XML/env_mach_specific.py

```diff
@@ -54,44 +54,43 @@ def populate(self, machobj):
                 for node in nodes:
                     self.add_child(node)
 
-    def _get_modules_for_case(self, compiler, debug, mpilib):
+    def _get_modules_for_case(self, case):
         module_nodes = self.get_nodes("modules")
         modules_to_load = None
         if module_nodes is not None:
-            modules_to_load = self._compute_module_actions(module_nodes, compiler, debug, mpilib)
+            modules_to_load = self._compute_module_actions(module_nodes, case)
 
         return modules_to_load
 
-    def _get_envs_for_case(self, compiler, debug, mpilib):
+    def _get_envs_for_case(self, case):
         env_nodes = self.get_nodes("environment_variables")
 
         envs_to_set = None
         if env_nodes is not None:
-            envs_to_set = self._compute_env_actions(env_nodes, compiler, debug, mpilib)
+            envs_to_set = self._compute_env_actions(env_nodes, case)
 
         return envs_to_set
 
-    def load_env(self, compiler, debug, mpilib):
+    def load_env(self, case):
         """
         Should only be called by case.load_env
         """
         # Do the modules so we can refer to env vars set by the modules
         # in the environment_variables block
-        modules_to_load = self._get_modules_for_case(compiler, debug, mpilib)
+        modules_to_load = self._get_modules_for_case(case)
         if (modules_to_load is not None):
             self.load_modules(modules_to_load)
 
-        envs_to_set = self._get_envs_for_case(compiler, debug, mpilib)
+        envs_to_set = self._get_envs_for_case(case)
         if (envs_to_set is not None):
             self.load_envs(envs_to_set)
 
+        self._get_resources_for_case(case)
 
-        self._get_resources_for_case(compiler, debug, mpilib)
-
-    def _get_resources_for_case(self, compiler, debug, mpilib):
+    def _get_resources_for_case(self, case):
         resource_nodes = self.get_nodes("resource_limits")
         if resource_nodes is not None:
-            nodes = self._compute_resource_actions(resource_nodes, compiler, debug, mpilib)
+            nodes = self._compute_resource_actions(resource_nodes, case)
```

Comment:
Changes to this file are worth a close review.

Reply:
This is what allows config_machines.xml to have any case variable as an attribute, looks good.